src/scene/shader-lib/chunks/common/vert/transform.js

```diff
@@ -56,10 +56,21 @@ mat4 getModelMatrix() {
     return matrix_model;
     #endif
 }
-
+#ifdef DISPLACEMENT
+uniform sampler2D texture_heightMap;
+uniform float material_displacementFactor;
+uniform float material_displacementOffset;
+#endif
 vec4 getPosition() {
     dModelMatrix = getModelMatrix();
+
```

Comment:
I'd leave this original line on top:
```
vec3 localPos = vertex_position;
```

and then add the displacement this way
```
#ifdef DISPLACEMENT
...
localPos == vertex_normal * displacer;
#endif
```